qiskit/transpiler/passes/optimization/hoare_opt.py

```diff
@@ -232,23 +232,24 @@ def _traverse_dag(self, dag):
 
             self._add_postconditions(gate, ctrl_ones, trgtqb, trgtvar)
 
-    def _target_successive_seq(self, qubit):
-        """gates are target successive if they have the same set of target
-            qubits and follow each other immediately on these target qubits
-            (consider sequences of length 2 for now)
+    def _remove_successive_identity(self, dag, qubit, max_idx=None):
+        """remove gates that have the same set of target qubits, follow each
+            other immediately on these target qubits, and combine to the
+            identity (consider sequences of length 2 for now)
         Args:
+            dag (DAGCircuit): the directed acyclic graph to run on.
             qubit (Qubit): qubit cache to inspect
-        Returns:
-            list(list(DAGOpNode or DAGOutNode)): list of target successive gate sequences for
-                                 this qubit's cache
+            max_idx (int): a value indicates a recursive call, optimize
+                           and remove gates up to this point in the cache
```

Comment:
This documentation is a bit confusing now - `_target_successive_seq` doesn't call itself recursively.

Reply:
I see your point here: the moved code/comment should make sense in the new context. I have updated the docstring. In addition, `max_idx` is a bit misleading here, so I renamed it to `from_idx`.